src/main/java/com/github/dockerjava/core/DockerClientConfig.java

```diff
@@ -102,8 +102,9 @@ private URI checkDockerHostScheme(URI dockerHost) {
     private String checkDockerCertPath(boolean dockerTlsVerify, String dockerCertPath) {
         if (dockerTlsVerify) {
             if (StringUtils.isEmpty(dockerCertPath)) {
-                throw new DockerClientException(
-                        "Enabled TLS verification (DOCKER_TLS_VERIFY=1) but certifate path (DOCKER_CERT_PATH) is not defined.");
+                return dockerCertPath;
+//                throw new DockerClientException(
```

Comment:
Maybe the existing config builder could create the SSLContext from dockerCertPath by default and pass it to the constructor of the config when build() is called. Then we could introduce a withCustomSSLContext method that overrides the default behavior. I think this would be the simplest and clean solution.